parsing/location.ml

```diff
@@ -459,20 +459,24 @@ let highlight_quote ppf
         (* Single-line error *)
         Format.fprintf ppf "%s | %s@," line_nb line;
         Format.fprintf ppf "%*s   " (String.length line_nb) "";
-        String.iteri (fun i c ->
+        (* Iterate up to [rightmost], which can be bigger than the length of
+           the line because it can span the newline character, which is not
+           included in [line]. Here it is guaranteed that [leftmost] and
+           [rightmost] are on the same line. *)
```

Comment:
I find this comment slightly confusing: `rightmost` can point to after the end of the line because the `Did you forget a ";"` message points to a location after the last token on the line. The fact that this means pointing to a newline character feels more coincidental than anything.

Maybe:
```
 (* Iterate up to [rightmost], which can be larger than the length of
     the line because we may point to a location after the end of the last token on the line, for instance
      {[
        token
                   ^
       Did you forget ...
       ]}
```
would be clearer?
(I am not sure if the reminder that we are in the single-line case is useful?)

Reply:
The reminder about single-line was to make sure that I wasn't accidentally talking about multi-line spans. Your version is better.